src/setting_sbp_cb.c

```diff
@@ -0,0 +1,401 @@
+                                       msg_settings_write_resp_t *write_response,
+                                       uint8_t len)
+{
+  settings_api_t *api = &ctx->client_iface;
+  if (api->send(api->ctx, SBP_MSG_SETTINGS_WRITE_RESP, len, (uint8_t *)write_response) != 0) {
+    api->log(log_err, "sending settings write response failed");
+    return -1;
+  }
+  return 0;
+}
+
+static void setting_update_value(settings_t *ctx, uint8_t *msg, uint8_t len)
+{
+  /* Extract parameters from message:
+   * 4 null terminated strings: section, name, value and type.
+   * Expect to find at least section, name and value.
+   */
+  const char *section = NULL, *name = NULL, *value = NULL, *type = NULL;
+  if (settings_parse((char *)msg, len, &section, &name, &value, &type) < SETTINGS_TOKENS_VALUE) {
+    ctx->client_iface.log(log_warning, "settings register resp cb, error parsing setting");
+    return;
+  }
+
+  /* Look up setting data */
+  setting_data_t *setting_data = setting_data_lookup(ctx->setting_data_list, section, name);
+  if (setting_data == NULL) {
+    return;
+  }
+
+  if (setting_data->watchonly) {
+    return;
+  }
+
+  settings_write_res_t write_result = setting_data_update_value(setting_data, value);
+
+  uint8_t resp[SETTINGS_BUFLEN] = {0};
+  uint8_t resp_len = 0;
+  msg_settings_write_resp_t *write_response = (msg_settings_write_resp_t *)resp;
+  write_response->status = write_result;
+  resp_len += sizeof(write_response->status);
+  int l = setting_data_format(setting_data,
+                              false,
+                              write_response->setting,
+                              sizeof(resp) - resp_len,
+                              NULL);
+  if (l < 0) {
+    return;
+  }
+  resp_len += l;
+
+  setting_send_write_response(ctx, write_response, resp_len);
+}
+
+static void setting_register_resp_callback(uint16_t sender_id, uint8_t len, uint8_t *msg, void *context)
+{
+  settings_t *ctx = (settings_t *)context;
+  (void)sender_id;
+
+  if (sender_id != SBP_SENDER_ID) {
+    ctx->client_iface.log(log_warning, "invalid sender %d != %d", sender_id, SBP_SENDER_ID);
+    return;
+  }
+
+  msg_settings_register_resp_t *resp = (msg_settings_register_resp_t *)msg;
+
+  if (resp->status == SETTINGS_REG_PARSE_FAILED) {
+    /* In case the request was corrupted during transfer, let the timeout trigger
+     * and request be sent again, parse error is printed in sbp_settings_daemon */
+    return;
+  }
+
+  /* Check for a response to a pending registration request */
+  int state = request_state_check(&ctx->request_state, &ctx->client_iface, resp->setting, len - sizeof(resp->status));
+
+  if (state > 0) {
+    /* No pending registration request */
+    return;
+  }
+
+  if (state < 0) {
+    ctx->client_iface.log(log_warning, "settings register resp cb, register req/resp mismatch");
+    return;
+  }
+
+  setting_update_value(ctx, (uint8_t *)resp->setting, len - sizeof(resp->status));
```

Comment:
we should only call this in case of a default value returned, so this could work:
```suggestion
  if (resp->status == SETTINGS_REG_OK_PERM) {
    setting_update_value(ctx, (uint8_t *)resp->setting, len - sizeof(resp->status));
  }
```